### ISSUE TITLE
Batch requests

### ISSUE THREAD

**Issue by Kitanotori** [microsoftgraph/msgraph-sdk-java#227]
### Expected behavior

Possibility to do batch requests (particularly posting data such as calendar events in batch).

Reference: https://docs.microsoft.com/en-us/graph/json-batching

### Actual behavior

No (obvious) way to do batch requests.

### Notes

Although the batch API supports batching requests of different types, personally I would be satisfied if the client supported just batching of similar requests (such as calendar event POSTs).

I would like to be able to do batch POSTs in the same way as single POSTs, like this:

`IGraphServiceClient::me().calendars(id).events().buildRequest().post(eventsList)`;

**Comment by NakulSabharwal** [comment]
Hi @Kitanotori , Thanks for creatign this issue. We are currently working on [msgraph-sdk-java](https://github.com/microsoftgraph/msgraph-sdk-java) and [msgraph-sdk-java-core](https://github.com/microsoftgraph/msgraph-sdk-java-core) integration. Since , batching has been added in Core https://github.com/microsoftgraph/msgraph-sdk-java-core/wiki/Batching it will come along with this library also after integration.

**Comment by dknchris** [comment]
Any updates on this? This is a much needed feature.

**Comment by NakulSabharwal** [comment]
@Kitanotori and @dknchris We have integrated [msgraph-sdk-core](https://github.com/microsoftgraph/msgraph-sdk-java-core) with [msgraph-sdk-java](https://github.com/microsoftgraph/msgraph-sdk-java) sdk and are now supporting batch request after version - 1.5.0. Thanks. Usage steps https://github.com/microsoftgraph/msgraph-sdk-java-core/wiki.

**Comment by Kitanotori** [comment]
@NakulSabharwal How do I make batch request using GraphServiceClient?

**Comment by tbelaire** [comment]
I too am interested in making batch requests using GraphServiceClient.

I looked at the linked instructions, but couldn't [find a reference](https://github.com/microsoftgraph/msgraph-sdk-java/search?q=MSBatchRequestStep&unscoped_q=MSBatchRequestStep) to MSBatchRequestStep in this repo.  Is the integration finished?

I don't want to use the OkHttpClient, as I just finished making a IHttpProvider wrapper around the Http client I want to use.
